### PR TITLE
Fix leader info repopulation

### PR DESCRIPTION
**Pull request by jefferai**
* Change how we populate and clear leader UUID. This fixes a case where
if a standby disconnects from an active node and reconnects, without the
active node restarting, the UUID doesn't change so triggers on a new
active node don't get run.

* Add a bunch of test helpers and minor updates to things.

### CODE REVIEW

File: vault/ha.go [hunk 1]
```diff
@@ -403,13 +418,22 @@ func (c *Core) waitForLeadership(newLeaderCh chan func(), manualStepDownCh, stop
 		if leaderLostCh == nil {
 			return
 		}
+
+		if atomic.LoadUint32(c.neverBecomeActive) == 1 {
+			c.heldHALock = nil
+			lock.Unlock()
+			c.logger.Info("marked never become active, giving up after interrupting perf standbys")
```
Comment (briankassouf) [inline vault/ha.go]: Is this log true? Seems like we are not interrupting here

File: vault/ha.go [hunk 2]
```diff
@@ -108,28 +108,41 @@ func (c *Core) Leader() (isLeader bool, leaderAddr, clusterAddr string, err erro
 		return false, "", "", nil
 	}
 
-	c.clusterLeaderParamsLock.RLock()
-	localLeaderUUID := c.clusterLeaderUUID
```
Comment (briankassouf) [inline vault/ha.go]: Can we now remove these values from Core?

File: vault/ha.go (hunk 1 above)
Comment (jefferai) [reply vault/ha.go]: Ah, it got moved. Will fix.

Comment (jefferai) [reply vault/ha.go]:
```suggestion
			c.logger.Info("marked never become active, giving up active state")
```

File: vault/ha.go (hunk 2 above)
Comment (jefferai) [reply vault/ha.go]: Ah yeah

**Pull request merged**